language/move-prover/stackless-bytecode-generator/src/dataflow_analysis.rs

```diff
@@ -22,24 +22,25 @@ pub trait AbstractDomain: Clone + Sized {
 }
 
 #[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
-pub struct BlockState<State: Clone> {
+pub struct BlockState<State: Clone, AnalysisError: Clone> {
```

Comment:
It would be likely better we just make the GlobalEnv accessible for analysis. Then you can generate errors as well as access all kind of other potentially needed information.